src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -34,29 +56,35 @@ class CommonReferenceRecording(BasePreprocessor):
         list of channels to be applied to each group is expected. If "single" reference, a list of one channel  or an
         int is expected.
     local_radius: tuple(int, int)
-        Use in the local CAR implementation as the selecting annulus (exclude radius, include radius)
-    verbose: bool
-        If True, output is verbose
+        Use in the local CAR implementation as the selecting annulus with the following format:
+
+        `(exclude radius, include radius)`
+
+        Where the exlude radius is the inner radius of the annulus and the include radius is the outer radius of the
+        annulus. The exclude radius is used to exclude channels that are too close to the reference channel and the
+        include radius delineates the outer boundary of the annulus whose role is to exclude channels
+        that are too far away.
+
     dtype: None or dtype
```

Comment:
I figure you might be interested in this debate. On neo we were discussing that dtype is not the real type. It is in fact a 
`numpy.typing.DTypeLike`, but that requires being up to date with your numpy version and it actually super confusing for the end-user although clearer for the developer. I like just calling it dtype because I think most people understand that dtype means whatever can be accepted by the `numpy.dtype()` constructor, but it is something to keep in mind if we ever thing about typing analysis.

Reply:
I think that type of very strict typing at this moment in python is more useful for people with application code.  A lot of the push for typing comes from people who mainly build apps (see fast api for example) and control the user envrionment to the degree that is impossible for us in library code. In the case of apps a typing as a schema / formal specification makes sense.

Library code like spikeinterface is very hard to type to that level. We can move the needled to add some typing so it helps the IDEs of users and works as an aid to documentation and that's kind of the goal to me.  From that perspective adding the more precise (DtypeLike) but at the same time more obscure typing does not make sense yet. 

In two three years one minimal is beyond 3.10 and we have better generics, the typing of array interfaces (e.g. numpy, cuda, dash, etc) is more worked out and spikeinterface data model stabilizes maybe we can start thinking on typing as a specification. Right now, I think we should prioritize clarify over formality.

For that reason I dislike abstracting away types away, that is defining somewhere else and then importig. The user should be able to see when they look at the signature what familiar stuff they can pass an expect.

Reply:
You’re preaching to the choir. That was the side of the discussion I took and the side we settled on for those exact reasons. But I was curious of your opinions. 

Thanks :)

Reply:
I am happy we agree. Can you point out to the discussion neo out of curosity?

Reply:
Sure [it's here](https://github.com/NeuralEnsemble/python-neo/pull/1334). It's spread every where, but the gist was that Neo doesn't currently have plans for type-checking so instead of being super strict with developer/abstract types that are strictly correct, that user facing helpful types would be better.